File: arrays_string/Palindrome_permutation/sol1.py
def palin_perm(input_str):
  count_dict = {}
  for s in input_str:
   if s == " ":
     continue
   elif s.lower() in count_dict:
     count_dict[s.lower()] += 1
   else:
     count_dict[s.lower()] = 1
  odd = False
  for s in count_dict:
    if count_dict[s] % 2 == 1:
      if odd:
        return False
      else:
        odd = True
  return True
  
  
  
if __name__ == "__main__":
  test_list = ["Tact Coa", "Tact Coaw", "Tact wCoaw"]
  
  for t in test_list:
    print t
    print palin_perm(t)

File: arrays_string/Palindrome_permutation/sol1.cpp
#include <ctype.h>
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>

bool Palinedrome_permutation(const std::string& str) {
  int char_a = 'a';
  unsigned marker = 0;
  for(auto& s : str) {
    if (s == ' ') {
      continue;
    }
    int c = tolower(s);
    marker ^= 1 << (c - char_a);
  }
  
  return marker == 0 || !( marker & marker - 1);
}

int main() {
  std::vector<std::string> test_list{"Tact Coa", "Tact Coaw", "Tact wCoaw"};
  
  for(auto& s : test_list) {
    std::cout << s << " ";
    if (Palinedrome_permutation(s)) {
      std::cout << "True";
    } else {
      std::cout << "False";
    }
    std::cout << std::endl;
  }
  return 0;
}